extension/zpagesextension/zpagesextension.go

```diff
@@ -69,12 +91,27 @@ func (zpe *zpagesExtension) Shutdown(context.Context) error {
 	if zpe.stopCh != nil {
 		<-zpe.stopCh
 	}
+
+	// Try to unregister the zpages span processor with the UnregisterSpanProcessor method.
+	// https://pkg.go.dev/go.opentelemetry.io/otel/sdk/trace#TracerProvider.UnregisterSpanProcessor.
+	// Use an anonymous interface to support non-SDK trace providers.
+	sdktracer, ok := zpe.telemetry.TracerProvider.(interface {
+		UnregisterSpanProcessor(SpanProcessor trace.SpanProcessor)
+	})
```

Comment:
nit: I would actually define a private interfaces instead of anonymous? Maybe also check for both funcs from the beginning?

Reply:
I did a single private interface https://github.com/open-telemetry/opentelemetry-collector/blob/9ac886315100151251823a89705ca4c43f3a2f4c/extension/zpagesextension/zpagesextension.go#L41-L54